Blockzilla/BrowserViewController.swift

```diff
@@ -358,6 +363,33 @@ class BrowserViewController: UIViewController {
 
         Telemetry.default.recordEvent(category: TelemetryEventCategory.action, method: TelemetryEventMethod.click, object: TelemetryEventObject.eraseButton)
     }
+    
+    private func requestReviewIfNecessary() {
+        guard #available(iOS 10.3, *) else { return }
+
+        let currentLaunchCount = UserDefaults.standard.integer(forKey: BrowserViewController.userDefaultsLaunchCountKey)
+        let threshold = UserDefaults.standard.integer(forKey: BrowserViewController.userDefaultsLaunchThresholdKey)
+
+        if threshold == 0 {
+            UserDefaults.standard.set(14, forKey: BrowserViewController.userDefaultsLaunchThresholdKey)
+        }
+
+        if currentLaunchCount <= threshold {
+            return
+        }
+
+        SKStoreReviewController.requestReview()
```

Comment:
is this async? can it go to the end of the function? That way all the threshold comparison and adjustment code happens first, then finally the UI code.